ebean-test/src/test/java/io/ebean/xtest/event/BeanPersistControllerTest.java

```diff
@@ -132,15 +134,124 @@ public void testInsertUpdateDelete_given_stopPersistingAdapter() {
     assertThat(stopPersistingAdapter.methodsCalled).containsExactly("preDeleteById");
     stopPersistingAdapter.methodsCalled.clear();
 
-    db.deleteAll(EBasicVer.class, Arrays.asList(22,23,24));
+    db.deleteAll(EBasicVer.class, Arrays.asList(22, 23, 24));
     assertThat(stopPersistingAdapter.methodsCalled).hasSize(3);
     assertThat(stopPersistingAdapter.methodsCalled).containsExactly("preDeleteById", "preDeleteById", "preDeleteById");
     stopPersistingAdapter.methodsCalled.clear();
 
     db.shutdown();
   }
 
-  private Database getDatabase(PersistAdapter persistAdapter) {
+  @Test
+  public void testCascade() {
+    Database db = getDatabase(new BeanPersistAdapter() {
+      @Override
+      public boolean isRegisterFor(Class<?> cls) {
+        return UTMaster.class == cls;
+      }
+
+      @Override
+      public boolean preDelete(BeanPersistRequest<?> request) {
+        return false;
+      }
+    });
+    Integer id;
+    UTMaster master = new UTMaster();
+    master.addDetail(new UTDetail());
+    db.save(master);
+    id = master.getId();
+
+    master = db.find(UTMaster.class, id);
+    assertThat(master.getDetails()).hasSize(1);
+
+    try (Transaction txn = db.beginTransaction()) {
+      txn.setBatchMode(true);
+      db.delete(master);
+      txn.commit();
+    }
+
+    master = db.find(UTMaster.class, id);
+    assertThat(master).isNotNull();
+    // CHECKME: Deleting of master was denied by the PersistListener
+    // What about detail? Is this intended, that it will be deleted?
+    assertThat(master.getDetails()).hasSize(0);
```

Comment:
CHECKME:
Can you tell me if the PersistController is intended to protect a bean from deletion? If so, then I would expect that if the deletion process was prevented by UtMaster, its details wouldn't be deleted.

But if you say the PersistController only takes care of the physical persistence, then that would be OK (although perhaps this should be explicitly mentioned again in the documentation for preInsert/Update/Delete).